strawberryfields/engine.py

```diff
@@ -277,15 +271,35 @@ def _broadcast_nones(val, dim):
                 p = p.compile(target, **compile_options)
             p.lock()
 
-            self._run_program(p, **kwargs)
-            shots = kwargs.get("shots", 1)
-            self.samples = [_broadcast_nones(p.reg_refs[k].val, shots) for k in sorted(p.reg_refs)]
+            _, values = self._run_program(p, **kwargs)
             self.run_progs.append(p)
 
+            if len(values) > 1:
+                sort_order = [
+                    r.ind for c in p.circuit for r in c.reg if "Measure" in c.op.__str__()
+                ]
```

Comment:
This could be turned into a nested list as per my previous comment:
```python
sort_order = [np.array([r.ind for r in c.reg]) for c in p.circuit if "Measure" in c.op.__str__()]
```

Reply:
both are very brief and great solutions! I'd personally suggest simply using `for` loops here so that each object can be well-named and readability could be increased